### ISSUE TITLE
Fast Enumerating ERC721s

### ISSUE THREAD

**Issue by klivin** [OpenZeppelin/openzeppelin-solidity#801]
## 🎉 Description

There Is no way to enumerate through someone that may have thousands of ERC721s

- [ ] 🐛 This is a bug report.
- [x] 📈 This is a feature request.

## 📝 Details

In regards to [ERC721Token contract](https://github.com/OpenZeppelin/zeppelin-solidity/blob/master/contracts/token/ERC721/ERC721Token.sol):

Want to add in subclass ERC721Token to allow more enumerable interface like shown in [this standard:](https://github.com/fulldecent/EIPs/blob/patch-2/EIPS/eip-721.md)

Namely I want to add the following in a ERC721Token instance:

```
  function tokenOfOwnerByIndex(address _owner, uint256 _index) 
    external 
    view 
    returns (uint256 _tokenId) 
  {
    require(_index < balanceOf(_owner));
    return ownedTokens[_owner][_index];
  }
```

This is an order of magnitude faster than using public tokensOf function which returns the entire array:
    return tokensOf(_owner)[_index];

However due to "private" encapsulation of ownedTokens in ERC721Token subclass:
  mapping (address => uint256[]) private ownedTokens;

We can't extend the functionality of that mapping, or any mapping in the base class ERC721.  To be more flexible, would it be crazy to make the mappings in the base class internal rather than private?

**Comment by spalladino** [comment]
Hey @klivin, for the next release of 721, we are working on both adding that method and changing the mappings to internal. So we hope to have both of those issues fixed. Keep an eye on https://github.com/OpenZeppelin/zeppelin-solidity/pull/682!

**Comment by elopio** [comment]
This was merged in #803. Please take a look and file a new issue if something is missing.
Thanks for the report @klivin !
